src/systems/acoustic_comms/AcousticComms.cc

```diff
@@ -61,8 +61,76 @@ class AcousticComms::Implementation
            std::tuple<std::chrono::duration<double>,
             std::chrono::duration<double>>>
           lastMsgReceivedInfo;
+
+  public: bool propagationModel(double _distToSource,
+                                int _numBytes);
+
+  /// \brief Source power in Watts.
+  /// \ref https://www.sciencedirect.com/topics/
+  /// computer-science/underwater-acoustic-signal
+  public: double sourcePower = 2000;
+
+  /// \brief Ratio of the noise intensity at the
+  /// receiver to the same reference intensity used for source level.
+  public: double noiseLevel = 1;
+
+  /// \brief Ratio of the total noise power at the array to the
+  /// noise received by the array along its main response axis.
+  /// \ref https://ieeexplore.ieee.org/document/5664178
+  public: double directivityIndex = 4;
+
+  /// \brief Information rate that can be transmitted over a given
+  /// bandwidth in a specific communication system, in (bits/sec)/Hz.
+  /// \ref: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5514747/
+  public: double spectralEfficiency = 7.0;
+
+  /// \brief Flag to store if the propagation model should be used.
+  public: bool usePropagationModel = false;
 };
 
+//////////////////////////////////////////////////
+bool AcousticComms::Implementation::propagationModel(
+  double _distToSource,
+  int _numBytes
+)
+{
+  // From https://www.mathworks.com/help/phased/ug/sonar-equation.html
+  // SNR = SL - TL - (NL - DI)
+  // SNR : Signal to noise ratio.
+  // SL : Source level. Ratio of the transmitted intensity from
+  //      the source to a reference intensity (1 m from source),
+  //      converted to dB.
+  // TL : Transmission loss (dB)
+  // NL : Noise level.
+  // DI : Receiver directivity index.
+
+  double sl = 171.5 + 10 * std::log10(this->sourcePower);
+  double tl = 20 * std::log10(_distToSource);
+
+  // Calculate SNR.
+  auto snr = sl - tl - (this->noiseLevel - this->directivityIndex);
```

Comment:
Doesn't the directivity index take into account the _direction_ of a sonar array? If we aren't modelling the directional response then we might as well account for this in our noise term.

Reply:
So do you mean the directivity index should default to zero (i.e. not modelling directional nature), unless specified ?

Reply:
I mean that unless we are factoring in direction to calculate the directivity index, it makes little sense to add this term.

Reply:
I see. Yeah I'll get rid of this term entirely.

Reply:
Removed : https://github.com/gazebosim/gz-sim/pull/1793/commits/1f39deff7c5084ab996d253bdca954bad6b40a44